extmod/uasyncio/core.py

```diff
@@ -68,65 +68,53 @@ def sleep(t):
 
 class IOQueue:
     def __init__(self):
-        self.poller = select.poll()
-        self.map = {}  # maps id(stream) to [task_waiting_read, task_waiting_write, stream]
+        import uevent
 
-    def _enqueue(self, s, idx):
-        if id(s) not in self.map:
-            entry = [None, None, s]
-            entry[idx] = cur_task
-            self.map[id(s)] = entry
-            self.poller.register(s, select.POLLIN if idx == 0 else select.POLLOUT)
+        self.poll = uevent.poll()
+
+    # This is not a method, just put here to keep the name "remove" out of global namespace.
+    def remove(self, task):
+        if self.data[0] is task:
+            idx = 0
+        elif self.data[1] is task:
+            idx = 1
+        elif self.data[2] is task:
+            idx = 2
         else:
-            sm = self.map[id(s)]
-            assert sm[idx] is None
-            assert sm[1 - idx] is not None
-            sm[idx] = cur_task
-            self.poller.modify(s, select.POLLIN | select.POLLOUT)
-        # Link task to this IOQueue so it can be removed if needed
-        cur_task.data = self
-
-    def _dequeue(self, s):
-        del self.map[id(s)]
-        self.poller.unregister(s)
+            return
+        self.data[idx] = None
+        self.unregister(1 << idx)
+
+    def _enqueue(self, s, idx):
```

Comment:
Yeah it's just the log2 of the event bit, so it's possible to index into the `data` list (I wanted to use small names here to reduce code size, because they allocate a qstr in the frozen code).